### PR TITLE
    Moved file stream events from outstream (os) to instream.pipe() (…

### PR DESCRIPTION
**Pull request by dbolack-ab**
…is.pipe()) as they were not triggering as written.

    'finished' event changed to 'finish' as this is the correct name.

    I am uncertain if this is a regression or node change or happenstance.

### CODE REVIEW

**Comment by pitaj** [comment]
From #5987 discussion:

I think a better solution for this is to pull out the copying here into a new `file.copyFile` function and use something like this solution instead: https://github.com/coolaj86/utile-fs/blob/master/fs.extra/fs.copy.js

It might be good to also use the node 8 `fs.copyFile` function if it is available.

File: src/file.js [hunk 1]
```diff
@@ -12,6 +12,59 @@ var utils = require('./utils');
...
 var file = module.exports;
 
+file.copyFile = function(src, dst, opts, cb) {
+	/* Wrapper for copyFile function with code to handle it not being present. If/When nodejs 8.5 or greater is a baseline requirement/assumption,
+	   this can be pruned and wholesale replace in all calls with fs.copyFile. */
+
+	function noop() {}
+
+  /* Test for availability of fs.CopyFile */
+	const semver = require('semver');
```
Comment (pitaj) [inline src/file.js]: Why not just use `typeof fs.copyFile !== 'function'`.

Also, you need to fix the formatting below. It doesn't match our linter at all.

File: src/file.js [hunk 2]
```diff
@@ -12,6 +12,58 @@ var utils = require('./utils');
...
+
+				fs.utimes(dst, stat.atime, stat.mtime, cb);
+			});
+		});
+	}
+
+	/* Test for availability of fs.CopyFile */
+	const semver = require('semver');
+
+	if (semver.gte(process.version, '8.5.0')) {
```
Comment (pitaj) [inline src/file.js]: Just use `typeof fs.copyFile !== 'function'` instead. No need for semver. It's safer to check if the function is there.

Comment (dbolack-ab) [reply src/file.js]: I hadn't thought of it that way. Good point. I'll make that change.

File: src/controllers/uploads.js [hunk 3]
```diff
@@ -228,12 +228,14 @@ function saveFileToLocal(uploadedFile, callback) {
...
 		function (upload, next) {
 			var storedFile = {
+				uid: uid,
 				url: nconf.get('relative_path') + upload.url,
 				path: upload.path,
 				name: uploadedFile.name,
 			};
-
-			plugins.fireHook('filter:uploadStored', { uploadedFile: uploadedFile, storedFile: storedFile }, next);
+			if (plugins.hasListeners('filter:uploadStored')) {
```
Comment (pitaj) [inline src/controllers/uploads.js]: I don't think this is necessary. If no listeners exist, it just returns what is passed in.

Comment (dbolack-ab) [reply src/controllers/uploads.js]: Okie dokie.

**Comment by pitaj** [comment]
Still a few issues here.

You need to explain why you're doing certain things when you do them.


- Why are you updating eslint? Let the core developers worry about that. It's what's causing the merge conflicts.
- Why are you adding a new `uid` parameter? Plugins might use these functions, have you considered that this might break them?
- As I understand it, the errors should be handled on both the input stream _and_ the output stream, as certain errors may not bubble through. Have you considered this?

Also please add tests for a few cases:
- An existing file is successfully copied
- Trying to copy a non-existent file results in an error
- Trying to copy to a file that already exists shouldn't result in an error

**Pull request closed without merging**